node/modules/chain.go

```diff
@@ -175,6 +231,68 @@ func NewSyncer(params SyncerParams) (*chain.Syncer, error) {
 	return syncer, nil
 }
 
+func NewSyncerFollower(params SyncerFollowerParams) (*chain.Syncer, error) {
+	var (
+		g  = params.GatewayAPI
+		lc = params.Lifecycle
+		sm = params.StateManager
+	)
+	lc.Append(fx.Hook{
+		OnStart: func(ctx context.Context) error {
+
+			headChanges, err := g.ChainNotify(ctx)
+			if err != nil {
+				return nil
+			}
+
+			go func() {
+				for {
+					head, ok := <-headChanges
+					if !ok {
+
+						//todo can we auto reconnect to the gateway?
+						//if the gateway disconnects is there a way to retry?
+						//panic is ok response because infrastructure should reboot rpc node upon error
+						// this will happen when the leader is restarted
+
+						log.Error("Notify stream was invalid")
+						panic("Notify stream was invalid")
```

Comment:
Yes, just call `g.ChainNotify(ctx)` in a loop with some sleep. go-jsonrpc will handle reconnecting under the hood, but the channel gets closed when the websocket jsonrpc connection drops.

Here's a good example - https://github.com/filecoin-project/lotus/blob/0374c5072804f5d8c79236944ef468af75f9627c/node/health.go#L48-L73